MT3-566 #ready-for-test Repeat the ID & residency check for each tenant

erbridge opened the pull request:
This is fairly massive, sorry.

It introduces a second database for process data, so we can loop through it sensibly, and that implied a bunch of related changes to support that.

We now add the current tenant's ID to the URL, and use that to work out which key to use for the database. This does mean that the general review section hook now needs some special casing for those repeating steps.

I forsee some issues when we add the tabination to the review page for each tenant (who reviews the answers for the absent tenants and how do we render that?).

erbridge committed:
Add support for nodes as cells in a table

erbridge committed:
Add support for repeating steps by modifying their paths

erbridge committed:
Be more reliable about parsing the URL for `Link`s

erbridge committed:
Capture tenant dates of birth

erbridge committed:
Add some basic styling to `Table`

erbridge committed:
Create a first pass verify step

This creates the step with the tenants in the table. We don't currently
loop through them all for the following steps, or use the answer to the
previous presence question.

erbridge committed:
Sort residents by id before storing them in the database

We want to be confident we're always handling them in the same order.

erbridge committed:
Forward the resident IDs to the next slugs for the id loop

erbridge committed:
Type the process data correctly to flatten some code

erbridge committed:
Pull the tenant data out into its own database and update the steps

This is a big commit, because a lot of things were fairly tightly
coupled to the concept of a single database for all process data, where
we now have two.

erbridge committed:
Fix the type of `renderValue`

erbridge committed:
Forward the process name to the client correctly

erbridge committed:
Switch to checking for the ends of URLs instead of the beginning

URLs might have a prefix, which we don't really care about.

erbridge committed:
Remove the tenant ids from the review page

We'll be displaying them in a different way.

LBHELewis commented on pages/verify.tsx:
```diff
@@ -0,0 +1,178 @@
+import Storage from "../storage/Storage";
+import tmpProcessRef from "../storage/processRef";
+
+export const SectionsPage: NextPage = () => {
```
Should this be `VerifyPage`?

erbridge replied:
Oops, yes.

LBHELewis commented on pages/verify.tsx:
```diff
@@ -0,0 +1,178 @@
+  );
+};
+
+export default SectionsPage;
```
Should this be `VerifyPage`?

erbridge commented on __tests__/features/prefetch-process-data.steps.ts:
```diff
@@ -44,7 +29,7 @@ defineFeature(loadFeature("./prefetch-process-data.feature"), test => {
   //   Given.iAmOffline(defineStep);
 
-  //   whenIStartTheProcess(defineStep);
+  //   When.iStartTheProcess(defineStep);
```
Probably not? I'll clear them.

erbridge committed:
Drop the unused offline tests

They don't work and there are no concrete plans on getting them working.

erbridge committed:
Drop all mention of online vs offline from feature tests

We don't test offline behaviour and do nothing to enure that we're
online, either, so it's misleading.

erbridge commented:
Please take another look @LBHELewis.
